src/dataflow/src/render/join.rs

```diff
@@ -277,4 +277,73 @@ where
             panic!("render_join called on invalid expression.")
         }
     }
+
+    pub fn render_semi_join(
```

Comment:
Comments here would help too! We might want to change the name; DD's `semijoin` is a generalization of the relational algebra semijoin to multisets, but I suspect most folks who use semijoins with SQL probably don't expect that.